infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationManager.java

```diff
@@ -204,4 +208,34 @@ public void processStandbyChange(TopologyDescriptor newConfig) {
             }
         }
     }
+
+    /**
+     * Stop the current log replication event and start a full snapshot sync for all standby clusters
+     */
+    public void enforceSnapshotFullSync() {
+        List<UUID> snapshotIds = new ArrayList<>();
+        for (CorfuLogReplicationRuntime standbyRuntime : runtimeToRemoteCluster.values()) {
+            standbyRuntime.getSourceManager().stopLogReplication();
+            UUID snapshotUUID = standbyRuntime.getSourceManager().startSnapshotSync();
+            snapshotIds.add(snapshotUUID);
+            log.info("Enforce Start full snapshot sync for cluster {} with uuid {}", standbyRuntime.getRemoteClusterId(), snapshotUUID);
+        }
+
+        // Block until the full snapshot sync has been started on all the standby clusters.
+        for (CorfuLogReplicationRuntime standbyRuntime : runtimeToRemoteCluster.values()) {
+            UUID snapshotUUID = null;
+            String clusterId = null;
+
+            try {
+                snapshotUUID = standbyRuntime.getSourceManager().getLogReplicationFSM().getSnapshotFullSyncUUID().get();
```

Comment:
the getSnapshotFullSyncUUID can happen before the SnapshotSyncRequest from line 219 is actually processed (as that just enqueues an event in the FSM). We could be holding on a previous CF which is not the one being instantiated in L290 of LogReplicationFSM.